src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -27,7 +27,7 @@ public unsafe partial class RaspberryPi3Driver : GpioDriver
         private const string DeviceTreeRanges = "/proc/device-tree/soc/ranges";
         private const string ModelFilePath = "/proc/device-tree/model";
 
-        private readonly IDictionary<int, PinMode> _sysFSModes = new Dictionary<int, PinMode>();
+        private readonly IDictionary<int, (PinMode CurrentPinMode, bool InUseBySysFs)> _pinModes = new Dictionary<int, (PinMode, bool)>();
```

Comment:
Now that I think of it, should this be a concurrentDictionary instead in case we are using it from multi-threaded scenarios?

Reply:
I replaced the dictionary with a fixed-size array. This is inherently thread-safe (as long as different pins are used, at least) and a lot faster than a dictionary.